File: backend/src/main/java/dev/datn/foco/service/OrderService.java
package dev.datn.foco.service;

import dev.datn.foco.dto.request.OrderRequest;
import dev.datn.foco.dto.respone.OrderResponse;

import java.time.LocalDateTime;
import java.util.List;

public interface OrderService {
    OrderResponse createOrder(OrderRequest request);
    
    OrderResponse getOrderById(Long id);
    
    OrderResponse getOrderByNumber(String orderNumber);
    
    List<OrderResponse> getAllOrders();
    
    List<OrderResponse> getOrdersByStore(Long storeId);
    
    List<OrderResponse> getOrdersByCustomer(Long customerId);
    
    List<OrderResponse> getOrdersByStatus(String status);
    
    List<OrderResponse> getOrdersByTable(Long tableId);
    
    List<OrderResponse> getOrdersByStoreAndStatus(Long storeId, String status);
    
    List<OrderResponse> getOrdersByDateRange(LocalDateTime startDate, LocalDateTime endDate);
    
    OrderResponse updateOrderStatus(Long orderId, String status);
    
    OrderResponse confirmOrder(Long orderId);
    
    OrderResponse completeOrder(Long orderId);
    
    OrderResponse cancelOrder(Long orderId);
    
    void deleteOrder(Long id);
}

File: frontend/src/services/orderService.js
import api from '../config/api';

const orderService = {
  getAllOrders: async () => {
    const response = await api.get('/orders');
    return response.data;
  },

  getOrderById: async (id) => {
    const response = await api.get(`/orders/${id}`);
    return response.data;
  },

  getOrdersByStore: async (storeId) => {
    const response = await api.get(`/orders/store/${storeId}`);
    return response.data;
  },

  getOrdersByStatus: async (status) => {
    const response = await api.get(`/orders/status/${status}`);
    return response.data;
  },

  createOrder: async (orderData) => {
    const response = await api.post('/orders', orderData);
    return response.data;
  },

  confirmOrder: async (id) => {
    const response = await api.patch(`/orders/${id}/confirm`);
    return response.data;
  },

  completeOrder: async (id) => {
    const response = await api.patch(`/orders/${id}/complete`);
    return response.data;
  },

  cancelOrder: async (id) => {
    const response = await api.patch(`/orders/${id}/cancel`);
    return response.data;
  },

  updateOrderStatus: async (id, status) => {
    const response = await api.patch(`/orders/${id}/status?status=${status}`);
    return response.data;
  },
};

export default orderService;

